src/Containers/Microsoft.NET.Build.Containers/AuthHandshakeMessageHandler.cs

```diff
@@ -143,27 +174,49 @@ private sealed record TokenResponse(string? token, string? access_token, int? ex
                 throw new ArgumentException(Resource.GetString(nameof(Strings.CouldntDeserializeJsonToken)));
             }
 
-            // save the retrieved token in the cache
-            var bearerAuth = new AuthenticationHeaderValue("Bearer", token.ResolvedToken);
-            return AuthHeaderCache.AddOrUpdate(realm, bearerAuth);
+            return new AuthenticationHeaderValue("Bearer", token.ResolvedToken);
         }
         else
         {
             return null;
         }
     }
 
+    private static async Task<DockerCredentials> GetLoginCredentials(string registry)
+    {
+        // For authentication with Docker Hub, 'docker login' uses 'https://index.docker.io/v1/' as the registry key.
+        // And 'podman login docker.io' uses 'docker.io'.
+        // Try the key used by 'docker' first, and then fall back to the regular case for 'podman'.
+        if (registry == ContainerHelpers.DockerRegistryAlias)
```

Comment:
This makes the tooling pick up credentials for Docker Hub that the user has provided by running the `docker login` command.

Reply:
Nice, we have part of this captured in https://github.com/dotnet/sdk-container-builds/issues/447 - you might have fixed this as well.

Reply:
It won't do _we should attempt a credential-less authentication_, but it will now pick up the credentials the user provided using the `docker login` command.